### PR TITLE
Use new URL filters

### LINKED ISSUE #122: Update template to use new `aboslute_url` and `relative_url` filters
So that we can DRY things up a bit. @pathawks since you're quickly becoming the resident expert, any interest?

### PR DESCRIPTION
**Pull request by pathawks**
I had to change some tests in addition to logic in this one, so let's give it a very close review.

Fixes #122

### CODE REVIEW

File: spec/jekyll_seo_tag_spec.rb [hunk 1]
```diff
@@ -218,18 +218,6 @@
...
-    let(:site) { make_site("github" => github_namespace) }
-
-    it "uses site.github.url to build the seo url" do
-      expected = %r!<link rel="canonical" href="http://example.invalid/page.html" \/>!
-      expect(output).to match(expected)
-      expected = %r!<meta property="og:url" content="http://example.invalid/page.html" />!
-      expect(output).to match(expected)
-    end
-  end
-
```
Comment (pathawks) [inline spec/jekyll_seo_tag_spec.rb]: Removed this whole test because my understanding is that this is being taken care of by jekyll/github-metadata#76?

Comment (benbalter) [inline spec/jekyll_seo_tag_spec.rb]: Yep. This plugin can now rely on `absolute_url`, which will be based on `site.url`, which Pages will set... no need to bake proprietary namespaces into open source plugins. :smile:

File: spec/jekyll_seo_tag_spec.rb [hunk 2]
```diff
@@ -449,8 +437,8 @@
     let(:context) { make_context({}, "paginator" => paginator) }
 
     it "outputs pagination links" do
-      expect(output).to match(%r!<link rel="prev" href="foo">!)
-      expect(output).to match(%r!<link rel="next" href="bar">!)
+      expect(output).to match(%r!<link rel="prev" href="/foo">!)
+      expect(output).to match(%r!<link rel="next" href="/bar">!)
```
Comment (benbalter) [inline spec/jekyll_seo_tag_spec.rb]: Duh. Glad we caught this.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -26,7 +26,7 @@ def make_page(options = {})
 end
 
 def make_post(options = {})
-  filename = File.expand_path("2015-01-01-post.md", CONFIG_DEFAULTS["source"])
+  filename = File.expand_path("_posts/2015-01-01-post.md", CONFIG_DEFAULTS["source"])
```
Comment (benbalter) [inline spec/spec_helper.rb]: No idea. https://github.com/jekyll/jekyll-seo-tag/commit/a42df44513b15a4d90b0c48afe5c8bddbac30cd2

Comment (pathawks) [inline spec/spec_helper.rb]: 😅

**Comment by pathawks** [comment]
@jekyllbot: merge +minor

**Comment by pathawks** [comment]
You've failed me, @jekyllbot

**Pull request merged**